### ISSUE TITLE
Improve the find/replace dialogue in the prepare menu

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#6534]
I am using an example dataset from the dslabs package.  @HawardKetoyoMsatsi this will be added to R-Instat.  Meantime add it yourself, from Tools >Add package.

Then I am doing work described in chapter 24 of the [Introduction to Data Science ](https://rafalab.github.io/dsbook/string-processing.html)book. 
This uses the dataset called `reported_heights`

There are various problems with this dataset as explained in chapter 24.  Here is a screenshot of the variable first - the heights are a character variable.
![image](https://user-images.githubusercontent.com/11226469/122032711-cd363580-cdc7-11eb-9ed1-6315e5829396.png)

Selecting the non-numeric rows shows those problems:

![image](https://user-images.githubusercontent.com/11226469/122033057-18504880-cdc8-11eb-9039-fed4678fff54.png)

The data were supposed to be entered in inches, but some people have given feet and inches.
The way the inches are given is actually more complicated than is shown above, see below:

The double-quotes are sometimes given as 2 single quotes.
 I would like to replace the dougle quotes (and also the 2 single quotes) by ok.  Just to show I can do this in R-Instat.

Here is the dialogue - using \ as the escape character.

![image](https://user-images.githubusercontent.com/11226469/122034777-c3adcd00-cdc9-11eb-8c8a-349fe70c90c4.png)

It works!

![image](https://user-images.githubusercontent.com/11226469/122034551-8d704d80-cdc9-11eb-9d3a-519b480257ec.png)

But when I return to the dialogue the string has been changed.  It should not be.  That's the correction to make:

And please check the other options in this dialogue.  I think they will have the same problem.

I think the double quote is the same, but the single quote is not.
![image](https://user-images.githubusercontent.com/11226469/122035266-35861680-cdca-11eb-9a84-910399b4ea33.png)

**Comment by HawardKetoyoMsatsi** [comment]
The dialog seems to remember what was input initially however there seems to be a problem with the way characters " . and ' are understood by the control. 
I tried to use . by replacing .. with xx 
![image](https://user-images.githubusercontent.com/16137042/123332623-67754680-d549-11eb-9047-02ad29a46e12.png)

look at what I get.
![image](https://user-images.githubusercontent.com/16137042/123332701-870c6f00-d549-11eb-9003-d69d264161b2.png)

Instead of looking for .. it takes any character it finds and replaces it with xx. 
@shadrackkibet kindly join this discussion and help me have  a look at how these characters (",' and .) are encoded.

**Comment by rdstern** [comment]
@HawardKetoyoMsatsi what is needed here is simple, but the bigger task is to help people with regex (regular expressions)  They are the default and you need to be aware of them.  For example "in regular expressions, the dot (.) is a special character used to match any one character."  That's why you got those changes.  If you wanted to include a dot, you need to type \\.

**Comment by HawardKetoyoMsatsi** [comment]
@rdstern @shadrackkibet @dannyparsons Something strange is happening with the String Handling dialog. dialog and the script window. 
I generated an R code from the dialog with one single quote (') and another with two single quotes (''). I later left without running anything. the script window accumulated many other single quotes now the code is messy.

The code I created
# Code generated by the dialog, String Handling

```
height <- data_book$get_columns_from_data(data_name="reported_heights", col_names="height", use_current_filter=FALSE)
count4 <- stringr::str_count(string=height, pattern="\'")
data_book$add_columns_to_data(data_name="reported_heights", col_name="count4", col_data=count4, before=FALSE, adjacent_column="height")

rm(list=c("count4", "height"))
# Code generated by the dialog, String Handling

height <- data_book$get_columns_from_data(data_name="reported_heights", col_names="height", use_current_filter=FALSE)
count4 <- stringr::str_count(string=height, pattern="\''")
data_book$add_columns_to_data(data_name="reported_heights", col_name="count4", col_data=count4, before=FALSE, adjacent_column="height")

rm(list=c("count4", "height"))
```
The code that created itself after leaving R-instat for some time without running anything.
```
# Code generated by the dialog, String Handling

height <- data_book$get_columns_from_data(data_name="reported_heights", col_names="height", use_current_filter=FALSE)
count4 <- stringr::str_count(string=height, pattern="\''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''")
data_book$add_columns_to_data(data_name="reported_heights", col_name="count4", col_data=count4, before=FALSE, adjacent_column="height")

rm(list=c("count4", "height"))
# Code generated by the dialog, String Handling

height <- data_book$get_columns_from_data(data_name="reported_heights", col_names="height", use_current_filter=FALSE)
count4 <- stringr::str_count(string=height, pattern="\''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''")
data_book$add_columns_to_data(data_name="reported_heights", col_name="count4", col_data=count4, before=FALSE, adjacent_column="height")

rm(list=c("count4", "height"))
```

**Comment by HawardKetoyoMsatsi** [comment]
@lloyddewit @rdstern @shadrackkibet  There is another funny behavior here where the Script window keeps updating itself as described in my above comment could you please have a look. I think as the dialog updates upon reopening that's where the extra quotes come from.

**Comment by rdstern** [comment]
@HawardKetoyoMsatsi I think there is some interesting regex behavior here.  I suggest this issue be raised by you at the start of the sprint.  Then perhaps you can work with one person to investigate further.  I am assuming that these dialogues will soon be improved anyway, so they become more powerful.  That's why I have been working on improving the regex facilities in the calculator recently.  I suggest that a part of the special regex keyboard will also be included in some of these dialogues and also that we include a checkbox option to turn regex off.  

This may develop into a bigger task, and hence be partly left for after the sprint, as we need lots of deliverables within the sprint.

**Comment by rdstern** [comment]
@HawardKetoyoMsatsi could you implement the first half of my suggestions, just above.  This should be very easy, will be useful in its own right, and will make the bug much rarer in use. 

That's because the bug is partly because the dialogues always use regex, and most of our users won't need that.  So we should turn it off as the default.

Then I suggest @Wycklife can add keyboards - so can do the remainder of my suggestions above.  

We should (later) still correct that bug, but please can we start with these changes.

**Comment by ChrisMarsh82** [comment]
fixed with pull request #6859
